packages/radix-vue/src/shared/useActiveElement.ts

```diff
@@ -0,0 +1,12 @@
+export function useActiveElement(): Element | null {
+  let activeElement = document.activeElement
+  if (activeElement == null) {
+    return null
+  }
+
+  while (activeElement != null && activeElement.shadowRoot != null && activeElement.shadowRoot.activeElement != null) {
+    activeElement = activeElement.shadowRoot.activeElement
+  }
+
+  return activeElement
+}

```

Comment:
This function will find the deepest active element regardeless of what shadowRoot it is in.